lang/java/reef-io/src/main/java/org/apache/reef/io/network/NetworkServiceClient.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.io.network;
+
+import org.apache.reef.exception.evaluator.NetworkException;
+import org.apache.reef.io.network.impl.DefaultNetworkServiceClientImpl;
+import org.apache.reef.tang.annotations.DefaultImplementation;
+import org.apache.reef.wake.EventHandler;
+import org.apache.reef.wake.Identifier;
+import org.apache.reef.wake.remote.Codec;
+import org.apache.reef.wake.remote.transport.LinkListener;
+
+@DefaultImplementation(DefaultNetworkServiceClientImpl.class)
+/**
+ * NetworkServiceClient.
+ *
+ * NetworkServiceClient is a service which is designed for communicating messages with each other.
+ * Users can send messages with NetworkServiceClient, by registering their Codec, EventHandler, and LinkListener.
+ */
+public interface NetworkServiceClient extends AutoCloseable {
```

Comment:
I agreed. Actually, I wanted to take this name as `NetworkService`. But, current `NetworkService` is an implementation class, not an interface. So I couldn't change this name to `NetworkService` because of compatibility. 

Well, it provides `ConnectionFactory`, so how about `NetworkConnectionFactoryProvider`?